src/training_raster_clipper/implementation/nbertaud.py

```diff
@@ -0,0 +1,194 @@
+
+    # Combine all the band in the same xarray
+    combined_data_array = xr.concat(combined_data_array, "band")
+
+    # Normalize the xarray
+    SPECIAL_VALUE = 0
+    RADIO_ADD_OFFSET = -1000
+    QUANTIFICATION_VALUE = 10000
+    normalized_data_array = combined_data_array.where(combined_data_array != SPECIAL_VALUE, np.nan)
+
+    normalized_data_array = (normalized_data_array + RADIO_ADD_OFFSET) / QUANTIFICATION_VALUE
+
+    return normalized_data_array
+
+
+def rasterize_geojson(
+    data_array: xr.DataArray,
+    training_classes: GeoDataFrame,
+) -> tuple[PolygonMask, FeatureClassNameToId]:
+    """Burns a set of vectorial polygons to a raster.
+
+    See https://gis.stackexchange.com/questions/316626/rasterio-features-rasterize
+
+    Args:
+        data_array (xr.DataArray): The Sentinel raster, from which data is taken, such as the transform or the shape.
+        training_classes (GeoDataFrame): The input set of classified multipolygons to burn
+
+    Returns:
+        xr.DataArray: A mask raster generated from the polygons, representing the same geographical region as the source dataarray param
+                      0 where no polygon were found, and integers representing classes in order of occurence in the GeoDataFrame
+    """
+
+    # Find transformation
+    transform = np.array(data_array.spatial_ref.GeoTransform.split(" "), dtype=float)
+    transform = rasterio.transform.Affine.from_gdal(*transform)
+
+    # Find array shape
+    array_shape = data_array.shape[1:]
+
+    # Regroup class if in the GeoJSON file there are multiple instances of the same class
+    dict_classes = {}
+    for id in training_classes.index:
+        current_class = training_classes["class"][id]
+
+        if not(current_class in dict_classes):
+            dict_classes[current_class] = []
+
+        dict_classes[current_class].append(training_classes["geometry"][id])
+
+    # Set a value for every class
+    geometry_shape = []
+    mapping = {}
+    for index, current_class in enumerate(dict_classes.items()):
+        mapping[current_class[0]] = index + 1
+        for geometry in current_class[1]:
+            geometry_shape.append((geometry, index + 1))
+
+    return rasterio.features.rasterize(geometry_shape, array_shape, 0, transform=transform), mapping
+
+
+def produce_clips(
+    data_array: xr.DataArray, burnt_polygons: PolygonMask, mapping: FeatureClassNameToId
+) -> ClassifiedSamples:
+    """Extract RGB values covered by classified polygons
+
+    Args:
+        data_array (xr.DataArray): RGB raster
+        burnt_polygons (PolygonMask): Rasterized classified multipolygons
+
+    Returns:
+        _type_: A list of the RGB values contained in the data_array and their corresponding classes
+    """
+    # Flatten list in (x,y) dimension
+    band_array = data_array.stack(flatten=("y", "x"))
+    class_array = xr.DataArray(burnt_polygons.reshape(-1), dims="flatten")
+
+    # Fusion the arrays
```

Comment:
This is not really a fusion but more a grouping
Now, these two variables share the same dimensions y and x. A Dataset is basically a dict of DataArrays sharing common dimensions